metricflow/specs.py

```diff
@@ -300,6 +302,15 @@ def bucket_hash(self) -> str:
         values.extend(sorted(self.window_groupings))
         return hash_strings(values)
 
+    @property
+    def as_time_dimension_spec(self) -> TimeDimensionSpec:  # noqa: D
+        return TimeDimensionSpec.from_name(self.name)
+
+    def __eq__(self, other: Any) -> bool:  # type: ignore[misc] # noqa: D
+        if not isinstance(other, NonAdditiveDimensionSpec):
+            return False
+        return self.bucket_hash == other.bucket_hash
```

Comment:
I was actually not aware of that! This shouldn't be needed then!

Reply:
ooo wait, just tested it, it doesn't actually work. It converts the values of the as a tuple in the order of the fields so it has nothing to do with the value itself. ie., if we had this
```
NonAdditiveDimensionSpec(
   name="asd", window_choice="min",window_groupings=(1,2)
)
```
it would convert it to 
```
("asd", "min", (1,2))
```
which would not be equal to say 
```
("asd", "min", (2,1))
```
which it should be equal